src/main/java/org/manifold/compiler/front/Main.java

```diff
@@ -214,16 +211,35 @@ public void reportContextSensitivity(@NotNull Parser recognizer, @NotNull DFA df
       throw new FrontendBuildException(errors.toString());
     }
 
-    ExpressionContextVisitor graphBuilder = new ExpressionContextVisitor();
+    ExpressionGraph exprGraph = new ExpressionGraph();
+
+    ImportVisitor importVisitor = new ImportVisitor();
+    importVisitor.visit(context);
+    for (String filePath : importVisitor.getImports()) {
+      File importedFile = new File(inputFile.getParent(), filePath);
+      ExpressionGraph g = parseFile(importedFile.getCanonicalFile());
+      exprGraph.addSubGraph(g);
+    }
+
+    ExpressionContextVisitor graphBuilder = new ExpressionContextVisitor(exprGraph);
     List<ExpressionContext> expressionContexts = context.expression();
     for (ExpressionContext expressionContext : expressionContexts) {
       graphBuilder.visit(expressionContext);
     }
-    ExpressionGraph exprGraph = graphBuilder.getExpressionGraph();
+    exprGraph = graphBuilder.getExpressionGraph();
+
     log.debug("writing out initial expression graph");
     File exprGraphDot = new File(inputFile.getName() + ".exprs.dot");
     exprGraph.writeDOTFile(exprGraphDot);
+    return exprGraph;
+  }
+
+  @Override
+  public Schematic invokeFrontend(CommandLine cmd) throws Exception {
+
+    File inputFile = Paths.get(cmd.getArgs()[0]).toFile();
 
+    ExpressionGraph exprGraph = parseFile(inputFile);
```

Comment:
What's the behavior in these cases?

1. An import references a file that does not exist.
2. A manifold file imports a file that is invalid and cannot have an `ExpressionGraph` built from it.

It might be worth adding fixtures in `buildErrors/` for these kinds of cases (particularly the first).